src/dbnode/storage/mediator.go

```diff
@@ -175,9 +176,16 @@ func (m *mediator) EnableFileOps() {
 
 func (m *mediator) Report() {
 	m.databaseBootstrapManager.Report()
-	m.databaseRepairer.Report()
 	m.databaseFileSystemManager.Report()
 	m.databaseColdFlushManager.Report()
+
+	m.Lock()
+	processes := append(make([]BackgroundProcess, 0, len(m.backgroundProcesses)), m.backgroundProcesses...)
```

Comment:
This will alloc on every `Report()`; is it necessary to take a copy of these vs just running within the lock?
```golang
m.Lock()
for _, process := range processes {
	process.Report()
}

m.Unlock()
```

Reply:
Guess that's just me being paranoid about some dynamically registered `process.Report` taking a long time to run under the lock; will change.